### PR TITLE
Feature/mob-4226 show users position on track scene

### PR DESCRIPTION
**Pull request by AleksanderWedrychowskiKarhoo**
## JIRA
[MOB-4226](https://karhoo.atlassian.net/browse/MOB-4226)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: KarhooUISDK/CommonUI/MapMVP/AppleMapKit/KarhooMKMapView.swift [hunk 1]
```diff
@@ -148,6 +149,22 @@ final class KarhooMKMapView: UIView, MapView, UIGestureRecognizerDelegate {
...
+        if let userLocation = CLLocationManager().location?.coordinate {
+            let viewRegion = MKCoordinateRegion(
+                center: userLocation,
+                latitudinalMeters: CLLocationDistance(standardZoom),
+                longitudinalMeters: CLLocationDistance(standardZoom)
+            )
+            mapView.setRegion(viewRegion, animated: true)
+            return true
+        } else {
+            assertionFailure()
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/MapMVP/AppleMapKit/KarhooMKMapView.swift:163]: This causes the app to crash instead of just returning false. Please remove it

**Review by dianakarh — CHANGES_REQUESTED** [review]
Extra failure assertion causes the SDK to crash

File: KarhooUISDK/CommonUI/MapMVP/AppleMapKit/KarhooMKMapView.swift [hunk 2]
```diff
@@ -247,22 +264,25 @@ extension KarhooMKMapView: MKMapViewDelegate {
...
-            let icon = UIImageView(image: iconImage)
-            icon.translatesAutoresizingMaskIntoConstraints = false
-            icon.contentMode = .scaleAspectFit
-            view.addSubview(icon)
-            
-            icon.centerX(inView: view)
-            icon.centerY(inView: view, constant: -6)
-            icon.anchor(width: 16, height: 16)
+        if annotation is MKUserLocation {
+            return nil
```
Comment (dianakarh) [inline KarhooUISDK/CommonUI/MapMVP/AppleMapKit/KarhooMKMapView.swift:268]: In what scenario does it pass through this line? I can't seem to find a case.

Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK/CommonUI/MapMVP/AppleMapKit/KarhooMKMapView.swift:268]: Default MKMapView user location marker is `MKUserLocation`.

File: KarhooUISDK/CommonUI/MapMVP/AppleMapKit/KarhooMKMapView.swift (hunk 1 above)
Comment (AleksanderWedrychowskiKarhoo) [reply KarhooUISDK/CommonUI/MapMVP/AppleMapKit/KarhooMKMapView.swift]: The crash'd to occur only on debug setup. Nonetheless, I removed it.

**Review by dianakarh — APPROVED** [review]

**Pull request merged**